### PR TITLE
Rasterize with the correct color component order.

### PR DESCRIPTION
**Pull request by bunnei**
- Fixes a regression with #594.

### CODE REVIEW

File: src/video_core/rasterizer.cpp [hunk 1]
```diff
@@ -48,10 +48,10 @@ static const Math::Vec4<u8> GetPixel(int x, int y) {
 
     u32 value = *(color_buffer_u32 + x + y * registers.framebuffer.GetWidth());
     Math::Vec4<u8> ret;
```
Comment (Subv) [inline src/video_core/rasterizer.cpp]: We'll need to add a switch to this function too, but imo that's better left for another PR.

**Comment by Subv** [comment]
LGTM

File: src/video_core/rasterizer.cpp [hunk 2]
```diff
@@ -29,7 +29,7 @@ static void DrawPixel(int x, int y, const Math::Vec4<u8>& color) {
     switch (registers.framebuffer.color_format) {
     case registers.framebuffer.RGBA8:
     {
-        u32 value = (color.a() << 24) | (color.r() << 16) | (color.g() << 8) | color.b();
+        u32 value = (color.r() << 24) | (color.g() << 16) | (color.b() << 8) | color.a();
```
Comment (neobrain) [inline src/video_core/rasterizer.cpp]: I think this code only will work correctly on little-endian CPUs. For better portability, each byte should be initialized individually.

File: src/video_core/rasterizer.cpp [hunk 3]
```diff
@@ -48,10 +48,10 @@ static const Math::Vec4<u8> GetPixel(int x, int y) {
 
     u32 value = *(color_buffer_u32 + x + y * registers.framebuffer.GetWidth());
     Math::Vec4<u8> ret;
-    ret.a() = value >> 24;
-    ret.r() = (value >> 16) & 0xFF;
-    ret.g() = (value >> 8) & 0xFF;
-    ret.b() = value & 0xFF;
+    ret.r() = value >> 24;
```
Comment (neobrain) [inline src/video_core/rasterizer.cpp]: Likewise, the color components here should not be extracted from a u32 here but from a u8 data stream.

**Pull request merged**